bentoml/_internal/frameworks/fastai.py

```diff
@@ -0,0 +1,299 @@
+
+
+if TYPE_CHECKING:
+    import fastai
+    from fastai.learner import Learner
+    from fastai.learner import load_learner  # type: ignore
+    from fastai.callback.core import Callback
+
+    from .. import external_typing as ext
+    from ..tag import Tag
+    from ...types import ModelSignature
+    from ..models.model import ModelSignaturesType
+
+try:
+    import torch
+    import torch.nn as nn
+except ImportError:  # pragma: no cover
+    raise MissingDependencyException(
+        "fastai requires `torch` as a dependency. Please follow PyTorch instruction at https://pytorch.org/ in order to use `fastai`."
+    )
+
+
+def _import_fastai() -> types.ModuleType:
+    try:
+        import fastai.basics  # type: ignore # noqa
+    except ImportError:  # pragma: no cover
+        raise MissingDependencyException("BentoML only supports fastai v2 onwards.")
+
+    try:
+        import fastai  # type: ignore
+    except ImportError:  # pragma: no cover
+        raise MissingDependencyException(
+            "fastai is required in order to use module 'bentoml.fastai'. Install fastai with 'pip install fastai'. For more information, refer to https://docs.fast.ai/#Installing."
+        )
+    return fastai
+
+
+__all__ = ["load_model", "save_model", "get_runnable", "get"]
+
+
+def get(tag_like: str | Tag) -> bentoml.Model:
+    """
+    Get the BentoML model with the given tag.
+
+    Args:
+        tag_like: The tag of the model to retrieve from the model store.
+
+    Returns:
+        :obj:`~bentoml.Model`: A BentoML :obj:`~bentoml.Model` with the matching tag.
+
+    Example:
+
+    .. code-block:: python
+
+       import bentoml
+       # target model must be from the BentoML model store
+       model = bentoml.fastai.get("fai_learner")
+    """
+    model = bentoml.models.get(tag_like)
+    if model.info.module not in (MODULE_NAME, __name__):
+        raise NotFound(
+            f"Model {model.tag} was saved with module {model.info.module}, failed to load with {MODULE_NAME}."
+        )
+    return model
+
+
+def load_model(bento_model: str | Tag | bentoml.Model) -> Learner:
+    """
+    Load the ``fastai.learner.Learner`` model instance with the given tag from the local BentoML model store.
+
+    If the model uses ``mixed_precision``, then the loaded model will also be converted to FP32. Learn more about `mixed precision <https://docs.fast.ai/callback.fp16.html>`_.
+
+    Args:
+        bento_model: Either the tag of the model to get from the store, or a BentoML `~bentoml.Model` instance to load the model from.
+
+    Returns:
+        :code:`fastai.learner.Learner`:
+            The :code:`fastai.learner.Learner` model instance loaded from the model store or BentoML :obj:`~bentoml.Model`.
+
+    Example:
+
+    .. code-block:: python
+
+       import bentoml
+
+       model = bentoml.fastai.load_model("fai_learner")
+       results = model.predict("some input")
+    """  # noqa
+
+    if not TYPE_CHECKING:
```

Comment:
is this check necessary?

Reply:
This is to trick pyright and not give up typing.

Reply:
Actually I realized we should just use our lazyloader at the top-level and that would solve everything...

Reply:
oo thats true.